### ISSUE TITLE
Brave browser launches but crashes immediately with segmentation fault

### ISSUE THREAD

**Issue by jayadevanraja** [brave/brave-browser#31753]
## Description 

When launching Brave browser from the terminal using either 
`/usr/bin/brave-browser`  
or   
`/snap/bin/brave`

Brave launches but immediately crashes with a segmentation fault error:

```
/usr/bin/brave-browser --incognito
[19923:19923:0720/153022.082073:ERROR:chrome_browser_cloud_management_controller.cc(162)] Cloud management controller initialization aborted as CBCM is not enabled.  
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

/usr/bin/brave-browser: line 48: 19923 Segmentation fault      (core dumped) "$HERE/brave" "$@"
```

I have tried:
- Reinstalling Brave via both the .deb and Snap packages  
- Deleting and recreating the Brave cache  
- Updating all Ubuntu packages

But the issue persists after each attempt.

System Info:
- Ubuntu 20.04
- Brave browser 1.56.9

## Steps to Reproduce

   1. In terminal, type "/usr/bin/brave-browser --incognito" or /snap/bin/brave --incognito, as the case may be.
   2. See the error messages

## Actual result:

Segmentation Fault

## Expected result:
Browser opens in incognito mode, as expected

## Reproduces how often: 

After today's (2023-07-20) update, Brave never opens in incognito mode.

## Version/Channel Information:

- Brave browser 1.56.9

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Not Tried
- Can you reproduce this issue with the nightly channel? Not Tried

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? Never Tried
- Does the issue resolve itself when disabling Brave Rewards? Never Tried
- Is the issue reproducible on the latest version of Chrome? Yes. Google Chrome crashes at "google-chrome --incognito". Chromium is in "Version 114.0.5735.198 (Official Build) snap (64-bit)", and it does not crash.

## Miscellaneous Information:

Brave started crashing only after current update. I believe this to be the same case with Google Chrome. Chromium "Version 114.0.5735.198 (Official Build) snap (64-bit)" does not crash, and that is the latest stable build available now for Ubuntu GNU/Linux system. So, I do not have access to version 117.0.5900.0 of Chromium to try it.

**Comment by nicolasfiel** [comment]
Hi, I have the same crash and error message since this update too.
Brave: Version 1.57.47 Chromium: 116.0.5845.96 (Build officiel)  (64 bits)
OS: Ubuntu 18.04.6 LTS (I know, old one)
The only way to open a new page in incognito mode without crashing is from a non-incognito page.
Direct launch of brave in incognito mode (from the menu or in command-line) crashes systematically.
Opening a new incognito page from an open incognito page crashes all brave processes.

**Comment by nicolasfiel** [comment]
I forgot I already had it enabled in fact, here are my 5 latest crash reports:
ID du rapport d'erreur importé : | 37e00100-0e1b-930a-0000-000000000000
Heure de l'importation : | jeudi 17 août 2023 à 20:14:11

ID du rapport d'erreur importé : | f8840000-0e1b-930a-0000-000000000000
Heure de l'importation : | jeudi 17 août 2023 à 00:18:44

ID du rapport d'erreur importé : | c1390000-0e1b-930a-0000-000000000000
Heure de l'importation : | mercredi 16 août 2023 à 20:10:15

ID du rapport d'erreur importé : | f95f0800-b0e3-890a-0000-000000000000
Heure de l'importation : | mardi 15 août 2023 à 09:55:50

ID du rapport d'erreur importé : | 0df60100-b0e3-890a-0000-000000000000
Heure de l'importation : | jeudi 10 août 2023 à 22:37:00

**Comment by jayadevanraja** [comment]
@wknapik  I am so sorry for the late reply.

Yes, the problem is still there, as seen from this terminal session:

```
appu@appu-LIFEBOOK-A555:~$ google-chrome --incognito
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

[271314:271314:0819/222646.834519:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: org.freedesktop.portal.Error.NotFound: Requested setting not found
Segmentation fault (core dumped)
appu@appu-LIFEBOOK-A555:~$ brave --incognito

Command 'brave' not found, but can be installed with:

sudo snap install brave

appu@appu-LIFEBOOK-A555:~$ brave-browser --incognito
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

[271616:271616:0819/222718.715329:ERROR:chrome_browser_cloud_management_controller.cc(162)] Cloud management controller initialization aborted as CBCM is not enabled.
/usr/bin/brave-browser: line 48: 271616 Segmentation fault      (core dumped) "$HERE/brave" "$@"
appu@appu-LIFEBOOK-A555:~$
appu@appu-LIFEBOOK-A555:~$ chromium-browser --incognito

Command 'chromium-browser' not found, but can be installed with:

sudo apt install chromium-browser

appu@appu-LIFEBOOK-A555:~$ chromium --incognito
ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (failed to map segment from shared object): ignored.
Gtk-Message: 22:32:55.958: Not loading module "atk-bridge": The functionality is provided by GTK natively. Please try to not load it.
[272492:272492:0819/223257.357895:ERROR:chrome_browser_cloud_management_controller.cc(163)] Cloud management controller initialization aborted as CBCM is not enabled.
[272492:272492:0819/223301.749524:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: org.freedesktop.portal.Error.NotFound: Requested setting not found
[0819/223302.520169:ERROR:ptracer.cc(567)] ptrace: Input/output error (5)
[0819/223302.640242:ERROR:elf_dynamic_array_reader.h(64)] tag not found
[0819/223302.684545:ERROR:elf_dynamic_array_reader.h(64)] tag not found
Segmentation fault (core dumped)
appu@appu-LIFEBOOK-A555:~$ 

```

This problem is existent for Chrome and Brave. Chromium also  has developed this issue now. I now have Chromium version "Version 116.0.5845.96 (Official Build) snap (64-bit)".

If I should report this elsewhere too, I shall do that. The problem is there only for chromium-based browsers, and not Firefox.

**Comment by jayadevanraja** [comment]
@wknapik I don't know how to do that. If this is what you mean, this is what I got in brave://crashes/

```
Show developer details
Crash from Saturday, 19 August 2023 at 22:27:19
Status:	Not uploaded
Local crash context:	e36fbec1-4ac1-4c29-8ed7-e960d2682d8a
```

If this is not what you meant, please say how to find the crash IDs.

**Comment by ivan-aksamentov** [comment]
Same issue for me. Also happens in Chromium.

Could you please relay this to chromium folks and/or teach me how to do it?


Crash from Sunday, August 20, 2023 at 2:27:52 AM
Status: Uploaded
Uploaded Crash Report ID: ce6b0500-0e1b-930a-0000-000000000000
Local Crash Context:  8d0349f1-ca34-491a-9c29-88c38fbef9f2
Upload Time:  Sunday, August 20, 2023 at 2:29:03 AM

Brave Version: 1.57.47 Chromium: 116.0.5845.96
Operating System: Ubuntu 20.04 (Linux 5.8.0-45-generic)

URL (if applicable) where crash occurred: on startup when adding `--incognito` flag

Can you reproduce this crash?

What steps will reproduce this crash? (If it's not reproducible, what were you doing just before the crash?)
1. Install Brave on Ubuntu 20.04 from [official Brave repos](https://brave.com/linux/)
2. Open terminal and run `brave-browser --incognito`
3. Observe window shows for a split second then disappers. Observe the following console output:
	```
	MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0
	
	[3068704:3068704:0820/022751.965065:ERROR:chrome_browser_cloud_management_controller.cc(163)] Cloud management controller initialization aborted as CBCM is not enabled.
	/usr/bin/brave-browser: line 48: 3068704 Segmentation fault      "$HERE/brave" "$@"
	[3068770:3068783:0820/022752.202137:ERROR:ssl_client_socket_impl.cc(968)] handshake failed; returned -1, SSL error code 1, net_error -3
	```

This only happens when starting Brave/Chromium in incognito mode. Opening it in normal mode works fine. Opening new incognito window from menu works fine. 

The same issue in Chromium since version 113-115 or so. I use Chromium from debian repos.

The only version of Chromium I could find and install that does not crash is 112:

```
apt-get install chromium=112.0.5615.138-1\~deb11u1 chromium-common=112.0.5615.138-1\~deb11u1 chromium-driver=112.0.5615.138-1\~deb11u1 libwebpmux3=0.6.1-2.1
```

(when downgrading, explicit version of `libwebpmux3` is required, because otherwise apt-get complains its incompatible)

The other versions of chromium I tried: 115, 116 - all crash. I could not find versions 113 and 114 to try.

The earliest currently available version of brave in the apt repo 1.51.118 still crashes and afew of the randomly picked versions between this and the latest version 1.57.47 all crash too.

****DO NOT CHANGE BELOW THIS LINE****
Crash ID: crash/ce6b0500-0e1b-930a-0000-000000000000

**Comment by jayadevanraja** [comment]
@wknapik If I use --incognito in the  GUI menu, the browsers (Brave, Chrome, and now Chromium also) do not show any crash reporting dialogue box for me. The icon comes in the Unity sidebar (I use Unity Desktop) and then it vanishes. If I use it in the terminal, the reason is shown as segmentation fault.

**Comment by xcariba** [comment]
Same issue. Never noticed it in neither chrome nor chromium. Brave version 1.84.141, Fedora 42. It appears in both incognito and regular mode. Freshly installed, I wanted to migrate from chrome, but it never worked more than couple of seconds.
Installed from official repos.
Regular launch looks like that. It says something about GPU and VAAPI (I use GNOME on wayland and nvidia RTX 3080 graphics card with proprietary driver).

```
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
[349340:349340:1119/221051.177525:ERROR:media/gpu/vaapi/vaapi_wrapper.cc:1631] vaInitialize failed: unknown libva error
[349340:349340:1119/221051.179909:ERROR:components/viz/service/main/viz_main_impl.cc:189] Exiting GPU process due to errors during initialization
[349295:349295:1119/221051.370137:ERROR:chrome/browser/ui/views/user_education/impl/browser_user_education_interface_impl.cc:154] Attempting to show IPH IPH_DiscardRing before browser initialization complete; IPH will not be shown.
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
[349454:349454:1119/221051.619340:ERROR:media/gpu/vaapi/vaapi_wrapper.cc:1631] vaInitialize failed: unknown libva error
[349454:349454:1119/221051.621433:ERROR:components/viz/service/main/viz_main_impl.cc:189] Exiting GPU process due to errors during initialization
[349488:349488:1119/221051.745836:ERROR:media/gpu/vaapi/vaapi_wrapper.cc:1631] vaInitialize failed: unknown libva error
/usr/bin/brave-browser: line 39: 349295 Segmentation fault      (core dumped) "$HERE/brave" "$@"
```

If I disable gpu acceleration using `--disable-gpu` it still exits in a few seconds.

```
[357455:357455:1119/221700.633462:ERROR:media/gpu/vaapi/vaapi_wrapper.cc:1631] vaInitialize failed: unknown libva error
[357411:357411:1119/221700.848900:ERROR:chrome/browser/ui/views/user_education/impl/browser_user_education_interface_impl.cc:154] Attempting to show IPH IPH_DiscardRing before browser initialization complete; IPH will not be shown.
terminate called without an active exception
/usr/bin/brave-browser: line 39: 357411 Aborted                 (core dumped) "$HERE/brave" "$@"
```

It usually prints all errors on start, except last line, then works for a few secnds and crashes with segfault.

**Comment by wknapik** [comment]
@xcariba this issue is 2.5 years old and for a xorg-based Ubuntu system. It's unlikely that the crashes you're seeing today on Wayland have any relation.

Chromium-based browsers are known to have issues with Wayland. I'm on xorg myself, but I've seen Wayland-related issues with workarounds including passing `--ozone-platform=wayland` and/or `--ozone-platform-hint=wayland`. You could give that a try. Starting with a fresh profile via something like `--user-data-dir=$(mktemp -d)` could eliminate some existing state as the cause.

**Comment by xcariba** [comment]
@wknapik Thanks! Unfortunately these workarounds did not work for me. I also tried `--ozone-platform=x11` and it crashed in the same way. I also had no problems with chrome.

**Comment by bsclifton** [comment]
Closing this particular issue as stale as there hasn't been an update to the root issue in 2.5 years

Let's open a new issue if there's something that is still happening @xcariba
